script.js

```diff
@@ -0,0 +1,167 @@
+const countriesGrid = document.querySelector('.countries-grid');
+let countryFetchedData = null;
+
+async function fetchCountriesData() {
+    if (countryFetchedData) {
+        return;
+    }
+    try {
+        const response = await fetch('./CountriesData.json');
+        if (!response.ok) {
+            throw `Failed to fetch countries data with status - ${response.status}`;
+        }
+        countryFetchedData = await response.json();
+    } catch (error) {
+        console.error('Error fetching countries data:', error);
+    }
+}
+
+function getDropdownsRegions(data) {
+    const dropdownRegionFilter = document.querySelector('.dropdown-body');
+    const uniqueRegionsSet = new Set(data.map(country => country.region));
+    const uniqueRegions = [...uniqueRegionsSet]
```

Comment:
In here you can  destruct your object, it is generally a good practice: 

`const uniqueRegionsSet = new Set(data.map(({ region }) => region));`